source/common/ssl/connection_impl.cc

```diff
@@ -139,16 +135,25 @@ Network::ConnectionImpl::PostIoAction ConnectionImpl::doHandshake() {
 
 void ConnectionImpl::drainErrorQueue() {
   bool saw_error = false;
+  bool saw_counted_error = false;
   while (uint64_t err = ERR_get_error()) {
     if (!saw_error) {
```

Comment:
I would just go with the simpler code (set always, which btw is almost definitely faster than the if check also since it will be on the stack and in cache, not that it matters)